src/frontend/screens/Settings/Experiments/BGMaps.tsx

```diff
@@ -0,0 +1,140 @@
+import * as React from "react";
+import { defineMessages, FormattedMessage, useIntl } from "react-intl";
+import { StyleSheet, Text, View } from "react-native";
+import { ScrollView, TouchableOpacity } from "react-native-gesture-handler";
+import { NavigationStackScreenComponent } from "react-navigation-stack";
+import MaterialIcon from "react-native-vector-icons/MaterialIcons";
+
+import { LIGHT_BLUE, LIGHT_GREY } from "../../../lib/styles";
+import HeaderTitle from "../../../sharedComponents/HeaderTitle";
+import { useExperiments } from "../../../hooks/useExperiments";
+import { useNavigation } from "react-navigation-hooks";
+
+const m = defineMessages({
+  BGMaps: {
+    id: "screens.Settings.Experiments.BGMaps.BGMaps",
+    defaultMessage: "Background Maps",
+  },
+  useBGMap: {
+    id: "screens.Settings.Experiments.BGMaps.useBGMap",
+    defaultMessage: "Use Background Map Feature",
+  },
+  goTo: {
+    id: "screens.Settings.Experiments.BGMaps.goTo",
+    defaultMessage: "Go to",
+  },
+});
+
+export const BGMapsSettings: NavigationStackScreenComponent = () => {
+  const [experiments, setExperiments] = useExperiments();
+
+  const { formatMessage: t } = useIntl();
+
+  const { navigate, goBack } = useNavigation();
+
+  return (
+    <ScrollView contentContainerStyle={styles.container}>
+      <Text style={[styles.header]}>
+        <FormattedMessage {...m.BGMaps} />
+      </Text>
+      <Text style={[{ marginBottom: 20 }, styles.text]}>
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+        tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim
+        veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea
+        commodo consequat. Duis aute irure dolor in reprehenderit in voluptate
+        velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint
+        occaecat cupidatat non proident, sunt in culpa qui officia deserunt
+        mollit anim id est laborum. Sed ut perspiciatis
+      </Text>
+
+      <Text style={[styles.text]}>
+        Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
+        tempor incididunt ut labore et dolore magna aliqua.
+      </Text>
+
+      <View style={[styles.switchContainer]}>
+        <Text style={[styles.text]}>
+          <FormattedMessage {...m.useBGMap} />
+        </Text>
+
+        <View
+          onTouchStart={() => {
+            setExperiments("backgroundMaps", !experiments.backgroundMaps);
+          }}
```

Comment:
I get an error that freezes the app when pressing this (infinite callstack when setting state in useEffect)